src/instructlab/process/process.py

```diff
@@ -0,0 +1,296 @@
+                sys.stderr = sys.__stderr__
+                process_registry.pop(local_uuid, None)
+                save_registry(process_registry=process_registry)
+
+
+def list_processes():
+    process_registry = load_registry()
+    if not process_registry:
+        logger.info("No processes currently in the registry.")
+        return
+
+    list_of_processes = []
+    for local_uuid, entry in process_registry.items():
+        now = datetime.now()
+
+        # Calculate runtime
+        runtime = now - datetime.fromisoformat(entry.get("start_time"))
+        # Convert timedelta to a human-readable string (HH:MM:SS)
+        total_seconds = int(runtime.total_seconds())
+        hours, remainder = divmod(total_seconds, 3600)
+        minutes, seconds = divmod(remainder, 60)
+
+        runtime_str = f"{hours:02}:{minutes:02}:{seconds:02}"
+        list_of_processes.append(
+            (
+                entry.get("type"),
+                entry.get("pid"),
+                local_uuid,
+                entry.get("log_file"),
+                runtime_str,
+            )
+        )
+
+    return list_of_processes
+
+
+def attach_process(local_uuid):
+    """
+    Attach to a running process and display its output in real-time.
+
+    Args:
+        local_uuid: UUID of the process to attach to
+    """
+    process_registry = load_registry()
+    if local_uuid not in process_registry:
+        logger.warning("Process not found.")
+        return
+
+    process_info = process_registry[local_uuid]
+    log_file = process_info["log_file"]
+
+    if not os.path.exists(log_file):
+        logger.warning(
+            "Log file not found. The process may not have started logging yet."
+        )
+        return
+
+    logger.info(
+        f"Attaching to process {local_uuid}. Press Ctrl+C to detach and Ctrl+D to kill."
+    )
+    all_pids = [process_info["pid"]] + process_info["children_pids"]
+    if not all_processes_running(all_pids):
+        return
+    try:
+        with open(log_file, "a+") as log:
+            log.seek(0, os.SEEK_END)  # Move to the end of the log file
+            while all_processes_running(all_pids):
+                line = log.readline()
+                # Check for non-empty and non-whitespace-only lines
+                if line.strip():
+                    print(line.strip())
+                else:
+                    time.sleep(0.1)  # Wait briefly before trying again
+    except KeyboardInterrupt:
+        logger.info("\nDetaching from and killing process.")
+    finally:
+        stop_process(local_uuid=local_uuid)
+
+
+def all_processes_running(pids: list[int]):
+    """
+    Returns if a process and all of its children are still running
+    Args:
+        pids (list): a list of all PIDs to check
+    """
+    return all(psutil.pid_exists(pid) for pid in pids)
+
+
+def stop_process(local_uuid):
+    """
+    Stop a running process.
+
+    Args:
+        uuid (int): uuid of the process to stop.
+    """
```

Comment:
Doesn't match function args, also I think `uuid` should be a string, not an int

Reply:
right, thanks